app/Console/Commands/CacheDelegateResignationIds.php

```diff
@@ -31,6 +31,9 @@ public function handle(): void
             ->select('sender_public_key', 'id')
             ->withScope(DelegateResignationScope::class)
             ->cursor()
-            ->each(fn ($transaction) => CacheResignationId::dispatch($transaction->sender_public_key, (string) $transaction->id)->onQueue('resignations'));
+            ->each(function ($transaction) {
+                // @phpstan-ignore-next-line
```

Comment:
Ignored because PHPStan reports an error that `$transaction->id` is already a string... but it's not, it's an integer and since we have strict types removing `(string)` would report an exception.

Reply:
isn't it assuming it's a string because of the `->select('sender_public_key', 'id')`? could we add an explicit type to the collection it returns so phpstan understands what `$transaction` is exactly instead of defaulting to `mixed` for it? I didn't get the string error when I removed the phpstan line, instead it doesn't seem to understand that an object is returned with two properties:

```
  Line   Console/Commands/CacheDelegateResignationIds.php
 ------ ------------------------------------------------------------
  35     Cannot access property $id on (int|string).
  35     Cannot access property $sender_public_key on (int|string).
  ```

Reply:
A bit poor choice of words on my part, let me explain in more detail:

If we add typehint that `$transaction` is a Transaction object, it will complain about "Casting something to string that's already a string". It will know that `$transaction` is of Transaction object, but Transaction model at the top has a model property declaration that `$id` is a string, and a `TransactionViewModel` has `id(): string` method. I could change that typehint and return type to `int`, but then the tests will fail:

```
App\ViewModels\TransactionViewModel::id(): Return value must be of type int, string returned
```

What I mean:

- [TransactionViewModel#id returning a string. Setting that to `int` fails some tests](https://github.com/ArdentHQ/arkscan/blob/develop/app/ViewModels/TransactionViewModel.php#L63)
- [Transaction model indicating that ID is `string`. Changing that to `int` fails static analysis because of `TransactionViewModel`](https://github.com/ArdentHQ/arkscan/blob/develop/app/Models/Transaction.php#L29)

If we remove `(string)` cast, static analysis will output:

```
36     Parameter #2 $transactionId of job class App\Jobs\CacheResignationId constructor expects string in App\Jobs\CacheResignationId::dispatch(), int given.
```

Because that job class expects an int.

Basically, TLDR is:

- If I switch everything down the line to `int` -- tests will fail that "string is expected", static analysis passes
- If I switch everything down the line to `string` -- some logical test will fail (` Failed asserting that 50 matches expected 153.`) and static analysis fails

I can add TODO to that phpstan-ignore to note that I should take a look at that because I don't want to go over why test fail when changing typehints in this PR that's dedicated to Laravel 9.